### ISSUE TITLE
The selected table cell's height of FlatTableUI when dragging the TableHeader

### ISSUE THREAD

**Issue by weimsh** [JFormDesigner/FlatLaf#236]
When I select one row in the Table and then select one column in the row(Now the column is focused or selected), then I start dragging the table header to change the column's order (or its index) to another, but find the selected cell's height is 1px less than the other columns' in the row.

Please Notice that My TableUI's settings are below:
Table.showHorizontalLines=true
Table.showVerticalLines=true

And the flatlaf's version is lastest 0.46.
[
![snapshot](https://user-images.githubusercontent.com/8486742/104469617-ac3c9f00-55f3-11eb-80ac-f55a5048de21.png)
](url)

**Comment by DevCharly** [comment]
Thx for reporting. I can reproduce it.

**Comment by weimsh** [comment]
Thanks!

**Comment by DevCharly** [comment]
Found the reason for the issue:
there is a horizontal grid line painted at the bottom of the selected dragged cell.

The problem is the order of painting:
- for dragged column, the cells are painted before the grid is painted
  (so the grid is painted over the cell if `Table.intercellSpacing` is `0,0`)
- for non-dragged columns, the grid is painted before the cells are painted
  (so the cell paints selection over the grid)

This painting is done in private methods in `BasicTableUI`.
Can't do much here (at the moment).

As workaround you could use following:
~~~java
UIManager.put( "Table.intercellSpacing", new Dimension( 0, 1 ) );
~~~

**Comment by DevCharly** [comment]
BTW while working on this issue, I noticed wrong grid line painting in dragged column on HiDPI screens:

![image](https://user-images.githubusercontent.com/5604048/107762404-43992d00-6d2d-11eb-8533-010e22e598c4.png)

This is fixed in commit 4f00591c4e59802a56aeeae7d283a42ab946b672
